### DISCUSSION TITLE
Runs without any UI!

### DISCUSSION THREAD

**Discussion by mbnoimi** [wwmm/easyeffects#2104] [Q&A] [answered]
I was using `PulseEffects` from APT. I deleted it using `apt purge` then installed flatpak version but after launching it. I unable to see any ui for it!!! Even if I launched it using terminal I get nothing!

```
$ /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=easyeffects-wrapper com.github.wwmm.easyeffects
^C
```
- Linux Mint 22.1 Xfce
- EasyEffects 7.0.0 (stable flatpak)
- Flatpak permissions
```ini
$ flatpak info --show-permissions com.github.wwmm.easyeffects
[Context]
shared=ipc;
sockets=x11;wayland;fallback-x11;
devices=dri;
filesystems=xdg-run/pipewire-0:ro;/home/admino/.themes;/usr/share/themes;

[Environment]
GTK_THEME=Mint-Y-Dark-Aqua
LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/app/lib/pipewire-0.3/jack
LV2_PATH=/app/lib/lv2:/app/extensions/Plugins/lv2
QT_STYLE_OVERRIDE=Adwaita-dark
```

**Comment by mbnoimi** [comment]
```
$ ps aux | grep easyeffects
admino     20626  0.4  0.0   3644  1064 pts/3    S+   11:45   0:00 bwrap --args 47 easyeffects-wrapper
admino     20636  0.4  0.0   3740  1580 pts/3    S+   11:45   0:00 bwrap --args 47 easyeffects-wrapper
admino     20637  0.0  0.0   7808  3760 pts/3    S+   11:45   0:00 /bin/bash /app/bin/easyeffects-wrapper
admino     20638  0.0  0.0   7808  1980 pts/3    S+   11:45   0:00 /bin/bash /app/bin/easyeffects-wrapper
admino     20676  0.0  0.0  10732  2364 pts/2    S+   11:45   0:00 grep --color=auto easyeffects
```
It doesn't work even if I killed the running processes and run it once again using
```
/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=easyeffects-wrapper com.github.wwmm.easyeffects
```

**Reply by wwmm** [reply]
As I di not user the Flatpak package I am not sure about what may be going on. Did you take a look at https://github.com/wwmm/easyeffects/issues/1802? Maybe it is a similar problem.

**Reply by wwmm** [reply]
Also try to run it in debug mode as explained in step number 3 https://github.com/wwmm/easyeffects/wiki/Reporting-Bugs

**Reply by mbnoimi** [reply]
> Also try to run it in debug mode as explained in step number 3 https://github.com/wwmm/easyeffects/wiki/Reporting-Bugs

Hell no! I used the 4th step ([flatpak method](https://github.com/wwmm/easyeffects/wiki/Reporting-Bugs#with-flatpak)) but I got:

```
$ flatpak-coredumpctl com.github.wwmm.easyeffects
Executable /app/delta/deltachat-desktop doesn't seem to be a flatpaked application.
Running: `"flatpak" "run" "--filesystem=home" "--filesystem=/tmp" "--command=gdb" "--devel" "com.github.wwmm.easyeffects" "/app/delta/deltachat-desktop" "/tmp/tmp0ahfq9jf"`
error: runtime/org.gnome.Sdk/x86_64/43 not installed
Traceback (most recent call last):
  File "/usr/bin/flatpak-coredumpctl", line 83, in <module>
    coredumper.run()
  File "/usr/bin/flatpak-coredumpctl", line 58, in run
    subprocess.check_call(flatpak_command)
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['flatpak', 'run', '--filesystem=home', '--filesystem=/tmp', '--command=gdb', '--devel', 'com.github.wwmm.easyeffects', '/app/delta/deltachat-desktop', '/tmp/tmp0ahfq9jf']' returned non-zero exit status 1.
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 153, in apport_excepthook
    with os.fdopen(os.open(pr_filename,
FileNotFoundError: [Errno 2] No such file or directory: '/var/crash/_usr_bin_flatpak-coredumpctl.1001.crash'

Original exception was:
Traceback (most recent call last):
  File "/usr/bin/flatpak-coredumpctl", line 83, in <module>
    coredumper.run()
  File "/usr/bin/flatpak-coredumpctl", line 58, in run
    subprocess.check_call(flatpak_command)
  File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['flatpak', 'run', '--filesystem=home', '--filesystem=/tmp', '--command=gdb', '--devel', 'com.github.wwmm.easyeffects', '/app/delta/deltachat-desktop', '/tmp/tmp0ahfq9jf']' returned non-zero exit status 1.
```

I haven't any  experience in building flatpaks packages but why `deltachat` interfere in this command!!!
BTW, `deltachat` works fine

**Reply by vchernin** [reply]
> I haven't any experience in building flatpaks packages but why `deltachat` interfere in this command!!! BTW, `deltachat` works fine

> `error: runtime/org.gnome.Sdk/x86_64/43 not installed`

The instructions were saying to install the wrong package, I updated it, please try follow the wiki again.

And please try launch the flatpak with 

`G_MESSAGES_DEBUG=easyeffects flatpak run com.github.wwmm.easyeffects`

**Reply by mbnoimi** [reply]
> `G_MESSAGES_DEBUG=easyeffects flatpak run com.github.wwmm.easyeffects`

It doesn't show anything although the processes are running (before running it I killed all the old processes)!
```
$ G_MESSAGES_DEBUG=easyeffects flatpak run com.github.wwmm.easyeffects

```
on another tab:
```
$ ps aux | grep easyeffects
admino    106742  0.0  0.0   3644  1124 pts/2    S+   22:36   0:00 bwrap --args 47 easyeffects-wrapper
admino    106752  0.0  0.0   3736  1584 pts/2    S+   22:36   0:00 bwrap --args 47 easyeffects-wrapper
admino    106753  0.0  0.0   7812  3704 pts/2    S+   22:36   0:00 /bin/bash /app/bin/easyeffects-wrapper
admino    106754  0.0  0.0   7812  1848 pts/2    S+   22:36   0:00 /bin/bash /app/bin/easyeffects-wrapper
admino    107011  0.0  0.0  10732  2280 pts/3    S+   22:44   0:00 grep --color=auto easyeffects
```

**Reply by wwmm** [reply]
>It doesn't show anything although the processes are running (before running it I killed all the old processes)!

But what was printed to the terminal while running it in debug mode? This may help to understand where the application is stopping. Unless EasyEffects is executed in service mode the window has to be shown.

**Reply by mbnoimi** [reply]
> But what was printed to the terminal while running it in debug mode? 

Nothing, it shows nothing!


https://user-images.githubusercontent.com/284547/215288050-f5455275-5ccd-4c1e-af4e-f37d93da1ecc.mp4

**Reply by wwmm** [reply]
>Nothing, it shows nothing!

There should be a lot of lines there. Even when an instance is already running in the background at least 3 or 4 lines should have been printed. Something may be happening to the executable wrapper being used by the Flatpak package.

**Reply by vchernin** [reply]
> Something may be happening to the executable wrapper being used by the Flatpak package.

The wrapper can be bypassed by running
``` 
G_MESSAGES_DEBUG=easyeffects flatpak run --command=easyeffects com.github.wwmm.easyeffects
```

**Reply by mbnoimi** [reply]
> ```
> G_MESSAGES_DEBUG=easyeffects flatpak run --command=easyeffects com.github.wwmm.easyeffects
> ```

```
$ G_MESSAGES_DEBUG=easyeffects flatpak run --command=easyeffects com.github.wwmm.easyeffects
(process:2): easyeffects-DEBUG: 00:49:07.694: 	easyeffects.cpp:35	easyeffects version: 7.0.0
(process:2): easyeffects-DEBUG: 00:49:07.702: 	easyeffects.cpp:45	locale directory: /app/share/locale
(easyeffects:2): easyeffects-DEBUG: 00:49:07.765: 	presets_manager.cpp:49	system input presets directory: "/app/etc/xdg/easyeffects/input"; 
(easyeffects:2): easyeffects-DEBUG: 00:49:07.765: 	presets_manager.cpp:49	system input presets directory: "/etc/easyeffects/input"; 
(easyeffects:2): easyeffects-DEBUG: 00:49:07.765: 	presets_manager.cpp:49	system input presets directory: "/etc/xdg/easyeffects/input"; 
(easyeffects:2): easyeffects-DEBUG: 00:49:07.765: 	presets_manager.cpp:52	system output presets directory: "/app/etc/xdg/easyeffects/output"; 
(easyeffects:2): easyeffects-DEBUG: 00:49:07.765: 	presets_manager.cpp:52	system output presets directory: "/etc/easyeffects/output"; 
(easyeffects:2): easyeffects-DEBUG: 00:49:07.765: 	presets_manager.cpp:52	system output presets directory: "/etc/xdg/easyeffects/output"; 
(easyeffects:2): easyeffects-DEBUG: 00:49:07.766: 	presets_manager.cpp:186user presets directory created: /home/admino/.var/app/com.github.wwmm.easyeffects/config/easyeffects/
(easyeffects:2): easyeffects-DEBUG: 00:49:07.766: 	presets_manager.cpp:186user presets directory created: /home/admino/.var/app/com.github.wwmm.easyeffects/config/easyeffects/input
(easyeffects:2): easyeffects-DEBUG: 00:49:07.766: 	presets_manager.cpp:186user presets directory created: /home/admino/.var/app/com.github.wwmm.easyeffects/config/easyeffects/output
(easyeffects:2): easyeffects-DEBUG: 00:49:07.767: 	presets_manager.cpp:186user presets directory created: /home/admino/.var/app/com.github.wwmm.easyeffects/config/easyeffects/autoload/input
(easyeffects:2): easyeffects-DEBUG: 00:49:07.767: 	presets_manager.cpp:186user presets directory created: /home/admino/.var/app/com.github.wwmm.easyeffects/config/easyeffects/autoload/output
(easyeffects:2): easyeffects-DEBUG: 00:49:08.400: 	pipe_manager.cpp:1426	compiled with PipeWire: 0.3.41
(easyeffects:2): easyeffects-DEBUG: 00:49:08.400: 	pipe_manager.cpp:1427	linked to PipeWire: 0.3.41
(easyeffects:2): easyeffects-DEBUG: 00:49:08.421: 	pipe_manager.cpp:1397	core version: 0.3.48
(easyeffects:2): easyeffects-DEBUG: 00:49:08.421: 	pipe_manager.cpp:1398	core name: pipewire-0
```

**Reply by wwmm** [reply]
After `core name: pipewire-0` there should be the following lines
```
pipe_manager.cpp:1311	found metadata: settings
pipe_manager.cpp:1311	found metadata: default
pipe_manager.cpp:1311	found metadata: route-settings
```
Usually when the default metadata is not available the reason is the package `pipewire-pulse` not being installed. But I am not sure this is the source of the problem here. Even when the default metadata is missing there are more lines in the logs.

**Reply by wwmm** [reply]
>I was using PulseEffects from APT. I deleted it using apt purge then installed flatpak version but after launching it. 

Are you sure you have PipeWire properly configured? I am not sure just installing the flatpak package will do this. As you were using Pulseaudio before you may have to do some PipeWire configuration before EasyEffects works.

**Reply by mbnoimi** [reply]
> Are you sure you have PipeWire properly configured?

What I've to do? All the apps using audio work fine.  
```
$ pactl info | grep "Server Name"
Server Name: pulseaudio
```

> As you were using Pulseaudio before you may have to do some PipeWire configuration before EasyEffects works.

May you please guide me what I've to do?

**Comment by mbnoimi** [comment, marked as answer]
Thank you guys.  I successfully enabled Pipewire using [this guide](https://github.com/mikeroyal/PipeWire-Guide#installing-pipewire-for-ubuntu) now I'm struggling EasyEffects itself because it refuses to keep on after closing the window although I applied rest permissions as mentioned in the message dialog!

![image](https://user-images.githubusercontent.com/284547/215307045-1332455c-db8d-4d48-b1e2-ee5631f6df8f.png)

```
flatpak permission-reset com.github.wwmm.easyeffects
```

Any way, I'll try to search in this forum about a solution for this issue.

**Reply by vchernin** [reply]
On Linux mint with xfce try install the gnome portal:

`sudo apt install --no-install-recommends xdg-desktop-portal-gnome`

Then reboot.

**Reply by mbnoimi** [reply]
Thank you. Works like a charm :)
